### ISSUE TITLE
$watch detects incorrectly a change on wire:click in Alpine 3.9.1

### ISSUE THREAD

**Issue by jamsouf** [alpinejs/alpine#2729]
### Discussed in https://github.com/alpinejs/alpine/discussions/2728

<div type='discussions-op-text'>

<sup>Originally posted by **jamsouf** March  5, 2022</sup>
Alpine 3.9.1 (in 3.9.0 this issue doesn't exist)
Livewire 2.10.4

I have an Alpine component inside a Livewire component. (Basically this is a Livewire triggered modal)
The `open` data should be watched and if it changed the Livewire event `fooEvent` should be called.

**In Alpine 3.9.0 this works as expected.**

```
<div>
    <div x-data="{ open: false }"
        x-show="open"
        x-init="$nextTick(() => { open = true; $watch('open', () => Livewire.emit('fooEvent')); })">
        ...
        ...
        <a href="#" wire:click.prevent="$set('foo', 'bar')">Click</a>
        <a href="#" @click="open = false">Close</a>
        ...
        ...
    </div>
</div>
```

**But the newest Alpine 3.9.1 breaked this code.**

If the link "Click" is pressed, the Livewire `$set()` method is correctly called, BUT also the Livewire event `fooEvent` is called.
`fooEvent` should not be called since the Alpine data `open` doesn't changed.</div>

**Comment by danharrin** [comment]
Can confirm, this is causing infinite watch loops after Livewire requests :(

I believe that 260679bec16f44c891b2c2082d020b53cc834eee might be something to do with it, but since there's no PR summarising these changes I'm not sure.

**Comment by danharrin** [comment]
Here's a simplified code example & screen recording of the Network tab:

```php
<div
    x-data="{ isOpen: false }"
    x-init="$watch('isOpen', () => $wire.foo = 'baz')"
>
    <button type="button" wire:click.prevent="$set('foo', 'bar')">
        Click
    </button>

    <p>
        Click the button, this should change to bar: {{ $foo }}
    </p>
</div>
```

Define `public $foo = '';` on the component class.

https://user-images.githubusercontent.com/41773797/159180002-dffd2c85-f5b2-4deb-9487-4f98f949f208.mp4

**Comment by calebporzio** [comment]
Thanks for providing that helpful recreation bit @danharrin.

This has been fixed. If you use Alpine v3.9.2 the infinite loop should stop.

More details for people's curiousity:

Here's the problem scenario:
- You register a $watch() somewhere inside a Livewire component, and inside that $watch you do something that triggers a Livewire request
- Normally the $watch will only fire when the thing it's watching changes
- After Alpine release 3.9.1, we stored the $watch property's reactive effect on the element it was declared on
- Livewire automatically runs all element reactive effects on every AJAX request
- This was causing the $watch effect (the callback) to run, causing another ajax request, and so on to infinity...
- The fix is to remove $watch's reactive effect form the element it was declared on, but still keep the "cleanup" attached to the element so that it will get tore down if the element was removed from the DOM

Does that make any sense? Probably not.. it's hard to explain. But anyhow, ITS FIXED!

Here's the commit to dig yourself: https://github.com/alpinejs/alpine/commit/d061f2d66011ab9ebeea8b4f21defb2e810ac624
